plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -133,25 +132,20 @@ func findNodesThatFit(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.No
 		fits := true
 		for name, predicate := range predicateFuncs {
 			fit, err := predicate(pod, node.Name, nodeNameToInfo[node.Name])
-			if err != nil {
-				switch e := err.(type) {
-				case *predicates.InsufficientResourceError:
-					if fit {
-						err := fmt.Errorf("got InsufficientResourceError: %v, but also fit='true' which is unexpected", e)
-						return api.NodeList{}, FailedPredicateMap{}, err
-					}
-				default:
-					return api.NodeList{}, FailedPredicateMap{}, err
-				}
+			if err != nil && fit {
+				// if we get a fit=true in case of a predicate error, we should just fail
+				// this would be an indication of a bug and should be addressed
+				// quick-fix would be to configure the scheduler to remove the predicate from the policy
+				err := fmt.Errorf("got error, but also fit='true' with predicate: %s, which is unexpected. Error: ", name, err.Error())
+				return api.NodeList{}, FailedPredicateMap{}, err
 			}
 			if !fit {
 				fits = false
 				if _, found := failedPredicateMap[node.Name]; !found {
 					failedPredicateMap[node.Name] = sets.String{}
 				}
-				if re, ok := err.(*predicates.InsufficientResourceError); ok {
-					failedPredicateMap[node.Name].Insert(re.Error())
-					break
+				if err != nil {
+					failedPredicateMap[node.Name].Insert(err.Error())
 				}
 				failedPredicateMap[node.Name].Insert(name)
```

Comment:
Sorry, I just noticed the map value is a set, not a string, so ignore previous question. But IIUC you've changed the behavior, such that in the original version of the code the set (value of failedPredicateMap) always had size one, but now it can have size two?